File: Micro Serialization Library (C#)/Components.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;

namespace MicroSerializationLibrary
{
	public class ImprovedSpinWait
	{

		public void SpinFor(double millisecondsTimeout)
		{
			SpinFor(Convert.ToInt64(millisecondsTimeout * TimeSpan.TicksPerMillisecond));
		}

		public void SpinFor(long Ticks)
		{
			Stopwatch s = new Stopwatch();
			s.Start();
#if NET20
			while (!(s.Elapsed.Ticks >= Ticks)) {
				System.Threading.Thread.SpinWait(1);
			}
#else
            while (s.Elapsed.Ticks >= Ticks) {
                System.Threading.Thread.SpinWait(16);
            }
			#endif
			s.Stop();
		}

	}
}
namespace MicroSerializationLibrary
{

	public enum MouseButton : short
	{
		Up = 0,
		Down = 1
	}
}

File: Micro Serialization Library/Components.vb
Public Class ImprovedSpinWait

    Public Overloads Shared Sub SpinFor(millisecondsTimeout As Double)
        SpinFor(CLng(millisecondsTimeout * TimeSpan.TicksPerMillisecond))
    End Sub

    Public Overloads Shared Sub SpinFor(Ticks As Long)
        Dim s As New Stopwatch
        s.Start()

        Do Until s.Elapsed.Ticks >= Ticks
            Threading.Thread.SpinWait(1)
        Loop

        s.Stop()
    End Sub

End Class

Public Enum MouseButton As Short
    Up = 0
    Down = 1
End Enum